pkg/controller/stacks/stack/stack_test.go

```diff
@@ -1688,6 +1701,657 @@ func Test_stackHandler_prepareHostAwarePodSpec(t *testing.T) {
 	}
 }
 
+type crdModifier func(*apiextensionsv1beta1.CustomResourceDefinition)
+
+func withCRDVersion(version string) crdModifier {
+	return func(c *apiextensionsv1beta1.CustomResourceDefinition) {
+		c.Spec.Version = version
+		c.Spec.Versions = append(c.Spec.Versions, apiextensionsv1beta1.CustomResourceDefinitionVersion{Name: version})
+	}
+}
+
+func withCRDScope(scope apiextensionsv1beta1.ResourceScope) crdModifier {
+	return func(c *apiextensionsv1beta1.CustomResourceDefinition) {
+		c.Spec.Scope = scope
+	}
+}
+
+func withCRDLabels(labels map[string]string) crdModifier {
+	return func(c *apiextensionsv1beta1.CustomResourceDefinition) {
+		meta.AddLabels(c, labels)
+	}
+}
+
+func withCRDSubresources() crdModifier {
+	return func(c *apiextensionsv1beta1.CustomResourceDefinition) {
+		c.Spec.Subresources = &apiextensionsv1beta1.CustomResourceSubresources{
+			Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			Scale:  &apiextensionsv1beta1.CustomResourceSubresourceScale{},
+		}
+	}
+}
+
+func withCRDGroupKind(group, kind string) crdModifier {
+	singular := strings.ToLower(kind)
+	plural := singular + "s"
+	list := kind + "List"
+
+	return func(c *apiextensionsv1beta1.CustomResourceDefinition) {
+		c.Spec.Group = group
+		c.Spec.Names.Kind = kind
+		c.Spec.Names.Plural = plural
+		c.Spec.Names.ListKind = list
+		c.Spec.Names.Singular = singular
+		c.SetName(plural + "." + group)
+	}
+}
+
+func crd(cm ...crdModifier) apiextensionsv1beta1.CustomResourceDefinition {
+	// basic crd with defaults
+	t := true
+	c := apiextensionsv1beta1.CustomResourceDefinition{
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Scope: "Namespaced",
+			Conversion: &apiextensionsv1beta1.CustomResourceConversion{
+				Strategy:                 apiextensionsv1beta1.NoneConverter,
+				WebhookClientConfig:      nil,
+				ConversionReviewVersions: nil,
+			},
+			PreserveUnknownFields: &t,
+		},
+	}
+	for _, m := range cm {
+		m(&c)
+	}
+	return c
+}
+
+type crModifier func(*rbac.ClusterRole)
+
+func withClusterRoleLabels(labels map[string]string) crModifier {
+	return func(cr *rbac.ClusterRole) {
+		meta.AddLabels(cr, labels)
+	}
+}
+
+func withClusterRoleRules(rules []rbac.PolicyRule) crModifier {
+	return func(cr *rbac.ClusterRole) {
+		cr.Rules = append(cr.Rules, rules...)
+
+	}
+}
+
+func clusterRole(name string, crm ...crModifier) rbac.ClusterRole {
+	c := rbac.ClusterRole{}
+	c.SetName(name)
+	for _, m := range crm {
+		m(&c)
+	}
+	return c
+}
+
+func Test_crdListFulfilled(t *testing.T) {
```

Comment:
lots of test cases and coverage in this file, thank you!!

Reply:
I think this latest commit may bump the coverage up even higher (by hitting the added code and code that was not previously covered due to the refactoring needed to placate the cyclomatic complexity of the stackinstall delete function).